### DISCUSSION TITLE
Getting error: ImportError: cannot import name 'Mapping' from 'collections'

### DISCUSSION THREAD

**Discussion by AmalNLal** [pyg-team/pytorch_geometric#7660] [Q&A]
Could anyone please help me out with this error. Here's the error trace:
ImportError                               Traceback (most recent call last)
Cell In[4], line 14
     11 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
     13 path = osp.join(osp.dirname(osp.realpath("test")), '../../data/MovieLens')
---> 14 dataset = MovieLens(path, model_name='all-MiniLM-L6-v2')
     15 data = dataset[0].to(device)
     17 # Add user node features for message passing:

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\torch_geometric\datasets\movie_lens.py:46, in MovieLens.__init__(self, root, transform, pre_transform, model_name)
     38 def __init__(
     39     self,
     40     root: str,
   (...)
     43     model_name: Optional[str] = 'all-MiniLM-L6-v2',
     44 ):
     45     self.model_name = model_name
---> 46     super().__init__(root, transform, pre_transform)
     47     self.data, self.slices = torch.load(self.processed_paths[0])

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\torch_geometric\data\in_memory_dataset.py:74, in InMemoryDataset.__init__(self, root, transform, pre_transform, pre_filter, log)
     66 def __init__(
     67     self,
     68     root: Optional[str] = None,
   (...)
     72     log: bool = True,
     73 ):
---> 74     super().__init__(root, transform, pre_transform, pre_filter, log)
     75     self._data = None
     76     self.slices = None

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\torch_geometric\data\dataset.py:102, in Dataset.__init__(self, root, transform, pre_transform, pre_filter, log)
     99     self._download()
    101 if self.has_process:
--> 102     self._process()

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\torch_geometric\data\dataset.py:235, in Dataset._process(self)
    232     print('Processing...', file=sys.stderr)
    234 makedirs(self.processed_dir)
--> 235 self.process()
    237 path = osp.join(self.processed_dir, 'pre_transform.pt')
    238 torch.save(_repr(self.pre_transform), path)

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\torch_geometric\datasets\movie_lens.py:67, in MovieLens.process(self)
     65 def process(self):
     66     import pandas as pd
---> 67     from sentence_transformers import SentenceTransformer
     69     data = HeteroData()
     71     df = pd.read_csv(self.raw_paths[0], index_col='movieId')

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\sentence_transformers\__init__.py:3
      1 __version__ = "2.2.2"
      2 __MODEL_HUB_ORGANIZATION__ = 'sentence-transformers'
----> 3 from .datasets import SentencesDataset, ParallelSentencesDataset
      4 from .LoggingHandler import LoggingHandler
      5 from .SentenceTransformer import SentenceTransformer

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\sentence_transformers\datasets\__init__.py:3
      1 from .DenoisingAutoEncoderDataset import DenoisingAutoEncoderDataset
      2 from .NoDuplicatesDataLoader import NoDuplicatesDataLoader
----> 3 from .ParallelSentencesDataset import ParallelSentencesDataset
      4 from .SentencesDataset import SentencesDataset
      5 from .SentenceLabelDataset import SentenceLabelDataset

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\sentence_transformers\datasets\ParallelSentencesDataset.py:4
      2 import logging
      3 import gzip
----> 4 from .. import SentenceTransformer
      5 from ..readers import InputExample
      6 from typing import List

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\sentence_transformers\SentenceTransformer.py:8
      6 from collections import OrderedDict
      7 from typing import List, Dict, Tuple, Iterable, Type, Union, Callable, Optional
----> 8 import requests
      9 import numpy as np
     10 from numpy import ndarray

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\requests\__init__.py:58
     55 except ImportError:
     56     pass
---> 58 from . import utils
     59 from .models import Request, Response, PreparedRequest
     60 from .api import request, get, head, post, patch, put, delete, options

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\requests\utils.py:26
     24 from . import __version__
     25 from . import certs
---> 26 from .compat import parse_http_list as _parse_list_header
     27 from .compat import (quote, urlparse, bytes, str, OrderedDict, unquote, is_py2,
     28                      builtin_str, getproxies, proxy_bypass, urlunparse,
     29                      basestring)
     30 from .cookies import RequestsCookieJar, cookiejar_from_dict

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\requests\compat.py:7
      1 # -*- coding: utf-8 -*-
      3 """
      4 pythoncompat
      5 """
----> 7 from .packages import chardet
      9 import sys
     11 # -------
     12 # Pythons
     13 # -------
     14 
     15 # Syntax sugar.

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\requests\packages\__init__.py:3
      1 from __future__ import absolute_import
----> 3 from . import urllib3

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\requests\packages\urllib3\__init__.py:10
      6 __license__ = 'MIT'
      7 __version__ = '1.10.4'
---> 10 from .connectionpool import (
     11     HTTPConnectionPool,
     12     HTTPSConnectionPool,
     13     connection_from_url
     14 )
     16 from . import exceptions
     17 from .filepost import encode_multipart_formdata

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\requests\packages\urllib3\connectionpool.py:38
     31 from .connection import (
     32     port_by_scheme,
     33     DummyConnection,
     34     HTTPConnection, HTTPSConnection, VerifiedHTTPSConnection,
     35     HTTPException, BaseSSLError, ConnectionError
     36 )
     37 from .request import RequestMethods
---> 38 from .response import HTTPResponse
     40 from .util.connection import is_connection_dropped
     41 from .util.retry import Retry

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\requests\packages\urllib3\response.py:9
      6 import io
      7 from socket import timeout as SocketTimeout
----> 9 from ._collections import HTTPHeaderDict
     10 from .exceptions import (
     11     ProtocolError, DecodeError, ReadTimeoutError, ResponseNotChunked
     12 )
     13 from .packages.six import string_types as basestring, binary_type, PY3

File ~\OneDrive\Desktop\Explainable A.I\Mini Project\.env\Lib\site-packages\requests\packages\urllib3\_collections.py:1
----> 1 from collections import Mapping, MutableMapping
      2 try:
      3     from threading import RLock

ImportError: cannot import name 'Mapping' from 'collections' (C:\Users\amall\AppData\Local\Programs\Python\Python311\Lib\collections\__init__.py)

**Comment by akihironitta** [comment]
Maybe running this will resolve your issue based on https://stackoverflow.com/questions/71632325/cannot-import-name-mapping-from-collections-on-importing-requests
```
pip install urllib3
```
